src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -247,11 +247,14 @@
           (.unlock lock))))))
 
 (defn maybe-check-for-updates
-  [config read-db]
+  [config read-db job-pool]
   (if (conf/foss? config)
-    (-> config
-        conf/update-server
-        (version/check-for-updates! read-db))
+    (let [checkin-interval-millis (* 1000 60 60 24)] ; once per day
+      (every checkin-interval-millis #(-> config
+                                          conf/update-server
+                                          (version/check-for-updates! read-db))
```

Comment:
That seems to be a side effect. The main intent is for products to check in to dujour on a more regular basis instead of only at startup (i.e. once per day).